data-pipeline-flink/user-cache-updater-2.0/src/main/scala/org/sunbird/dp/usercache/util/UserMetadataUpdater.scala

```diff
@@ -173,45 +158,70 @@ object UserMetadataUpdater {
   }
 
   def getCustodianUserInfo(metrics: Metrics, userId: String, config: UserCacheUpdaterConfigV2, cassandraConnect: CassandraUtil): mutable.Map[String, AnyRef] = {
-    val custodianExtUserMap = extractCustodianUserExternalData(readFromCassandra(
-      keyspace = config.keySpace,
-      table = config.userExternalIdTable,
-      clause = QueryBuilder.eq("userid", userId),
-      metrics,
-      cassandraConnect, config
-    ), config)
+    var userExternalInfo: mutable.Map[String, AnyRef] = mutable.Map[String, AnyRef]()
 
-    //fetch custodian User's channel from organisation table
-    val originalProvider = custodianExtUserMap.getOrElse("originalprovider", "")
-    val custChannel = getCustUserChannel(originalProvider, config, cassandraConnect, metrics)
-    custodianExtUserMap.+=(config.userChannelKey -> custChannel)
-  }
+    val userDeclarationInfo = readUserDeclaredInfo(userId, config, cassandraConnect, metrics)
+    if (null != userDeclarationInfo) {
+      metrics.incCounter(config.dbReadSuccessCount)
 
-  def extractCustodianUserExternalData(externalIdentity: util.List[Row], config: UserCacheUpdaterConfigV2): mutable.Map[String, AnyRef] = {
-    val result: mutable.Map[String, AnyRef] = mutable.Map[String, AnyRef]()
-    externalIdentity.forEach{row =>
-      row.getString("idtype").toLowerCase() match {
-        case config.declareExternalId => result.put(config.externalidKey, row.getString("externalid"))
-        case config.declaredSchoolName => result.put(config.schoolNameKey, row.getString("externalid"))
-        case config.declaredSchoolCode => result.put(config.schoolUdiseCodeKey, row.getString("externalid"))
-        case _ => result
+      val orgId = userDeclarationInfo.getString("orgid")
+      val validOrgId = validateOrgId(orgId, config, cassandraConnect, metrics)
```

Comment:
Why is the valid check for org required?

Reply:
In env like dev/staging we saw earlier that the organisation id (for us which is also a userchannel).. was not present.. we used to get defects related to that.
So with this implementation also.. i am keeping the same logic that if user’s orgid is not getting validated from organisation table.. we don’t provide its external information as well